runtime/parachains/src/configuration/migration.rs

```diff
@@ -0,0 +1,284 @@
+		pub hrmp_max_parachain_inbound_channels: u32,
+		pub hrmp_max_parathread_inbound_channels: u32,
+		pub hrmp_channel_max_message_size: u32,
+		pub code_retention_period: BlockNumber,
+		pub parathread_cores: u32,
+		pub parathread_retries: u32,
+		pub group_rotation_frequency: BlockNumber,
+		pub chain_availability_period: BlockNumber,
+		pub thread_availability_period: BlockNumber,
+		pub scheduling_lookahead: u32,
+		pub max_validators_per_core: Option<u32>,
+		pub max_validators: Option<u32>,
+		pub dispute_period: SessionIndex,
+		pub dispute_post_conclusion_acceptance_period: BlockNumber,
+		pub dispute_max_spam_slots: u32,
+		pub dispute_conclusion_by_time_out_period: BlockNumber,
+		pub no_show_slots: u32,
+		pub n_delay_tranches: u32,
+		pub zeroth_delay_tranche_width: u32,
+		pub needed_approvals: u32,
+		pub relay_vrf_modulo_samples: u32,
+	}
+
+	impl<BlockNumber: Default + From<u32>> Default for HostConfiguration<BlockNumber> {
+		fn default() -> Self {
+			HostConfiguration {
+				group_rotation_frequency: 1u32.into(),
+				chain_availability_period: 1u32.into(),
+				thread_availability_period: 1u32.into(),
+				no_show_slots: 1u32.into(),
+				validation_upgrade_frequency: Default::default(),
+				validation_upgrade_delay: Default::default(),
+				code_retention_period: Default::default(),
+				max_code_size: Default::default(),
+				max_pov_size: Default::default(),
+				max_head_data_size: Default::default(),
+				parathread_cores: Default::default(),
+				parathread_retries: Default::default(),
+				scheduling_lookahead: Default::default(),
+				max_validators_per_core: Default::default(),
+				max_validators: None,
+				dispute_period: 6,
+				dispute_post_conclusion_acceptance_period: 100.into(),
+				dispute_max_spam_slots: 2,
+				dispute_conclusion_by_time_out_period: 200.into(),
+				n_delay_tranches: Default::default(),
+				zeroth_delay_tranche_width: Default::default(),
+				needed_approvals: Default::default(),
+				relay_vrf_modulo_samples: Default::default(),
+				max_upward_queue_count: Default::default(),
+				max_upward_queue_size: Default::default(),
+				max_downward_message_size: Default::default(),
+				ump_service_total_weight: Default::default(),
+				max_upward_message_size: Default::default(),
+				max_upward_message_num_per_candidate: Default::default(),
+				hrmp_sender_deposit: Default::default(),
+				hrmp_recipient_deposit: Default::default(),
+				hrmp_channel_max_capacity: Default::default(),
+				hrmp_channel_max_total_size: Default::default(),
+				hrmp_max_parachain_inbound_channels: Default::default(),
+				hrmp_max_parathread_inbound_channels: Default::default(),
+				hrmp_channel_max_message_size: Default::default(),
+				hrmp_max_parachain_outbound_channels: Default::default(),
+				hrmp_max_parathread_outbound_channels: Default::default(),
+				hrmp_max_message_num_per_candidate: Default::default(),
+			}
+		}
+	}
+}
+
+fn migrate_to_v1<T: Config>() -> Weight {
```

Comment:
```suggestion

/// Migrates the `HostConfiguration` from v0 (with deprecated `hrmp_open_request_ttl` and without 
/// `ump_max_individual_weight`) to v1 (without hrmp ttl and with max individual weight).
/// Uses the `Default` implementation of `HostConfiguration` to choose a value for `ump_max_individual_weight`.
///
/// NOTE: Only use this function if you know what you are doing. Default to using `migrate_to_latest`.
pub fn migrate_to_v1<T: Config>() -> Weight {
```
Let's expose the function so users can decide which migration to run?

Reply:
Can you elaborate? I was under impression that the pallet owns it's data and performs it's migrations whenever needed and that can only be triggered on runtime upgrade.

Exposing this function will allow users to run the storage upgrade whenever, which... I do not know when that can be useful apart from a runtime upgrade.

Reply:
(I applied the rustdoc part in https://github.com/paritytech/polkadot/pull/3575/commits/7236e5e27a81cfaa8c1dcabdab661487f095f9aa)

Reply:
What you mention is the common case. Unfortunately it can happen that you cannot follow the common upgrade path and need to run a custom migration.

Maybe I'm thinking too far ahead/too much of the edge case :shrug: 
I just remember having to fork Substrate and change visibility of migrations to be able to trigger them in a custom upgrade scenario.

Reply:
That is enough of justification and I exposed the fields